### PR TITLE
Skip over broken TTF font when creating cache

### LINKED ISSUE #5232: RuntimeError: No SFNT name table
Hi, I'm on Ubuntu 14.04 LTS using Python 3.4 and matplotlib 1.4.3.

I'm new to all this , and I tried to make the simple lot in the tutorial (http://matplotlib.org/users/pyplot_tutorial.html), but I got these two errors:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/font_manager.py", line 1407, in <module>
    fontManager = pickle_load(_fmcache)
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/font_manager.py", line 960, in pickle_load
    with open(filename, 'rb') as fh:
FileNotFoundError: [Errno 2] No such file or directory: '/home/aamir/.cache/matplotlib/fontList.py3k.cache'




During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "testplots.py", line 1, in <module>
    import matplotlib.pyplot as plt
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/pyplot.py", line 27, in <module>
    import matplotlib.colorbar
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/colorbar.py", line 34, in <module>
    import matplotlib.collections as collections
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/collections.py", line 27, in <module>
    import matplotlib.backend_bases as backend_bases
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/backend_bases.py", line 56, in <module>
    import matplotlib.textpath as textpath
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/textpath.py", line 19, in <module>
    import matplotlib.font_manager as font_manager
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/font_manager.py", line 1415, in <module>
    _rebuild()
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/font_manager.py", line 1400, in _rebuild
    fontManager = FontManager()
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/font_manager.py", line 1055, in __init__
    self.ttflist = createFontList(self.ttffiles)
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/font_manager.py", line 589, in createFontList
    prop = ttfFontProperty(font)
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/font_manager.py", line 395, in ttfFontProperty
    sfnt = font.get_sfnt()
RuntimeError: No SFNT name table
```
-----
If I use Python2, I can get rid of the first FileNotFoundError, but I'd strongly prefer to use Python 3. Anyway, I have no idea what these errors mean. A bit of searching around lead to me believe I might have some incompatible fonts, but removing all the fonts from my ~/.fonts folder (which is where I installed all my custom fonts) didn't solve the issue. 

Anyone have any idea on what's going on?

### PR DESCRIPTION
**Pull request by mdboom**
Fixes #5232.

### CODE REVIEW

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -587,7 +587,7 @@ def createFontList(fontfiles, fontext='ttf'):
                 continue
             try:
                 prop = ttfFontProperty(font)
-            except KeyError:
+            except KeyError, RuntimeError:
```
Comment (tacaswell) [inline lib/matplotlib/font_manager.py]: python3 requires `()` on this.

Comment (mdboom) [inline lib/matplotlib/font_manager.py]: Doh!

**Comment by tacaswell** [comment]
back ported as 0cb18e0

**Comment by JGalvezTT** [comment]
I'm still getting the same error. When I try to 

##### import matplotlib.font_manager

I get an error in the function 

##### ttfFontProperty(Font)

on the line 396

**Comment by JGalvezTT** [comment]
I added ValueError at line 592.

##### except KeyError, RuntimeError:

**Comment by tacaswell** [comment]
@JGalvezTT What version of mpl are you using?

**Comment by JGalvezTT** [comment]
@tacaswell, I'm using 1.5.1 on Anaconda. I got it to work by adding ValueError as an exception on the line 592. But now I'm getting error in other libraries (Opencv). So it might be that something is wrong on my set up.

**Comment by dov** [comment]
I also got this error on fedora 24, matplotlib version 1.5.2rc2. The bad font was "Truetypewriter PolyglOTT" font available from: http://www.fontsaddict.com/font/truetypewriter-polyglott.html . The error trace was:

```
/usr/lib64/python2.7/site-packages/matplotlib/font_manager.py:273: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
Traceback (most recent call last):
  File "<stdin>", line 20, in <module>
  File "/usr/lib64/python2.7/site-packages/matplotlib/pyplot.py", line 29, in <module>
    import matplotlib.colorbar
  File "/usr/lib64/python2.7/site-packages/matplotlib/colorbar.py", line 34, in <module>
    import matplotlib.collections as collections
  File "/usr/lib64/python2.7/site-packages/matplotlib/collections.py", line 27, in <module>
    import matplotlib.backend_bases as backend_bases
  File "/usr/lib64/python2.7/site-packages/matplotlib/backend_bases.py", line 62, in <module>
    import matplotlib.textpath as textpath
  File "/usr/lib64/python2.7/site-packages/matplotlib/textpath.py", line 15, in <module>
    import matplotlib.font_manager as font_manager
  File "/usr/lib64/python2.7/site-packages/matplotlib/font_manager.py", line 1421, in <module>
    _rebuild()
  File "/usr/lib64/python2.7/site-packages/matplotlib/font_manager.py", line 1406, in _rebuild
    fontManager = FontManager()
  File "/usr/lib64/python2.7/site-packages/matplotlib/font_manager.py", line 1059, in __init__
    self.ttflist = createFontList(self.ttffiles)
  File "/usr/lib64/python2.7/site-packages/matplotlib/font_manager.py", line 590, in createFontList
    prop = ttfFontProperty(font)
  File "/usr/lib64/python2.7/site-packages/matplotlib/font_manager.py", line 396, in ttfFontProperty
    sfnt = font.get_sfnt()
ValueError: No SFNT name table
```

**Comment by tacaswell** [comment]
@dov can you make a new issue?  This fell through the cracks for 1.5.x :disappointed:

**Comment by dov** [comment]
Ok. I did. See: https://github.com/matplotlib/matplotlib/issues/6702 . Unfortunately I seem to have erased the font that caused the crash. :-(

**Pull request merged**